src/Enterspeed.Source.UmbracoCms.V7/Data/Repositories/EnterspeedJobRepository.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Enterspeed.Source.UmbracoCms.V7.Data.Models;
+using Enterspeed.Source.UmbracoCms.V7.Data.Schemas;
+using Umbraco.Core;
+
+namespace Enterspeed.Source.UmbracoCms.V7.Data.Repositories
+{
+    public class EnterspeedJobRepository
+    {
+        private readonly string _tableName = "EnterspeedJobs";
+        public IList<EnterspeedJob> GetFailedJobs()
+        {
+            var db = ApplicationContext.Current.DatabaseContext.Database;
+
+            var failedJobs = db.Fetch<EnterspeedJobSchema>(
+                $"SELECT * FROM [{_tableName}] WHERE [JobState] = {EnterspeedJobState.Failed.GetHashCode()} ORDER BY [CreatedAt]");
+
+            var result = failedJobs.Select(MapFromSchema).Where(x => x != null).ToList();
```

Comment:
Are we sure that db.Fetch<> cannot return null?

Reply:
It shouldn't, but I've updated it so it handles if it returns null